c10/core/ScalarTypeToTypeMeta.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <c10/core/ScalarType.h>
+#include <c10/util/typeid.h>
+
+namespace c10 {
+
+/**
+ * convert ScalarType enum values to TypeMeta handles
+ */
+static inline caffe2::TypeMeta scalarTypeToTypeMeta(ScalarType scalar_type) {
+#define DEFINE_CASE(ctype, name) \
+  case ScalarType::name:         \
+    return caffe2::TypeMeta::Make<ctype>();
+
+  switch (scalar_type) {
+    AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS(DEFINE_CASE)
+    case ScalarType::Undefined:
+      return caffe2::TypeMeta();
+    default:
+      AT_ERROR("Unrecognized Scalartype ", scalar_type, " (please report this error)");
+  }
+#undef DEFINE_CASE
+}
+
+/**
+ * convert TypeMeta handles to ScalarType enum values
+ * Caution: extremely hot
+ */
+static inline ScalarType typeMetaToScalarType(caffe2::TypeMeta dtype) {
```

Comment:
Re SIOF problems: would you bet the farm on this function never being called in a static initializer? I wouldn't...

Reply:
Yeah agree